src/cli/msig/index.ts

```diff
@@ -163,6 +166,61 @@ export default async function msig(args: string[]) {
                 newGovernanceAccounts
             );
 
+            if (test) await executeProposal(proposer, proposalName, proposalHash);
+        } else if (proposalType === 'deploy-contract') {
+            const contractName = args[3];
+
+            if (!contractName) {
+                throw new Error('Contract name must be provided for deploy-contract proposal');
+            }
+
+            const contractInfo = {
+                account: contractName,
+                contractDir: path.join(__dirname, `../../Tonomy-Contracts/contracts/${contractName}`),
+            };
+            const action = await deployContract(contractInfo, signer, { throughTonomyProxy: true, actionData: true });
+
+            if (action) {
+                const proposalHash = await createProposal(
+                    proposer,
+                    proposalName,
+                    [action],
+                    privateKey,
+                    newGovernanceAccounts
+                );
+
+                if (test) await executeProposal(proposer, proposalName, proposalHash);
+            }
+        } else if (proposalType === 'eosio.code') {
+            const contractAccount = 'vesting.tmy';
+            const permission = 'ops.tmy';
+
+            const action = {
+                account: 'eosio',
+                name: 'updateauth',
+                authorization: [
+                    {
+                        actor: contractAccount,
+                        permission: 'active',
+                    },
+                ],
+                data: {
+                    account: contractAccount,
+                    permission: 'owner',
+                    parent: '',
+                    auth: await updateControlByAccount(contractAccount, permission, signer, {
```

Comment:
updateControlByAccount() will call the blockchain, not return the Authority object